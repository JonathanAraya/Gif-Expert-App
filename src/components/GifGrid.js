import React, {Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
// import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([])
    const { loading } = useFetchGifs();

    // useEffect(() => {
    //     getGifs( category )
    //         .then( imgs => setImages( imgs ) )
    // }, [ category ]);

    return (
        <Fragment>
            <h3>{ category }</h3>
            {loading ? 'Cargando...' : 'Finalizado'}
            {/* <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
        </div> */}
        </Fragment>
    )
}
