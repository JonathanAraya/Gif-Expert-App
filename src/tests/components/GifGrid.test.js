import { shallow } from "enzyme"
import {GifGrid} from '../../components/GifGrid';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'Minecraft'

    test('Debe hacer match con el snapshot ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();

    });

    test('debe mostrar items luego de la respuesta del useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost.cl/img',
            title: 'Titulo'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();

    });
    
})
