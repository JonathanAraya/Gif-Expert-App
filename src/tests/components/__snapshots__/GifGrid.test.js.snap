// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe hacer match con el snapshot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Minecraft
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar items luego de la respuesta del useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Minecraft
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo"
      url="https://localhost.cl/img"
    />
  </div>
</Fragment>
`;
